Server/src/controllers/getBasicDataController.ts

```diff
@@ -0,0 +1,50 @@
+import { Request, Response } from 'express';
+import { retrieveBasicData } from '../services/getBasicDataService';
+
+export class getBasicDataController {
+    private retrieveBasicDataObject = new retrieveBasicData();
+    constructor() {
+    }
+
+    /**
+     * This controller will take a request, send it to the getDataService to acquire an array 
+     * containing the names and IDs all of the categories in the database
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForAllCategories(request: Request, response: Response) {

```

Comment:
These still need a try catch block because there is a chance that something will fail (unlikely this section of code will work 100% of the time). The question is where in the service is there a possibility that the code will fail. 

If you know exactly where the error can come from then we can throw an error inside the service and it can be caught in the controller and returned to the user.

Reply:
I see, that makes more sense yes. The one error source that comes to mind would occur when the model that the service is using/referencing cannot connect to the database.